qa/rpc-tests/receivedby.py

```diff
@@ -62,10 +62,22 @@ def run_test(self):
         assert_array_result(self.nodes[1].listreceivedbyaddress(11),{"address":addr},{ },True)
 
         #Empty Tx
-        addr = self.nodes[1].getnewaddress()
+        empty_addr = self.nodes[1].getnewaddress()
         assert_array_result(self.nodes[1].listreceivedbyaddress(0,True),
-                           {"address":addr},
-                           {"address":addr, "account":"", "amount":0, "confirmations":0, "txids":[]})
+                           {"address":empty_addr},
+                           {"address":empty_addr, "account":"", "amount":0, "confirmations":0, "txids":[]})
+
+        #Test Address filtering
```

Comment:
I'd prefer this test if `other_addr` (currently declared on line 78) received some funds before you did your tests on `listreceivedbyaddress()` with a filter address. This isn't currently testing the case where the wallet contains multiple addresses with funds and you want to see the transactions received by one of those addresses. Ideally the test case would be:

- define addr and send funds to it (already done for you in lines 42-43)
- define other_addr and send funds to it
- call `listreceivedbyaddress()` filtering on `addr`. Assert only one transaction is returned.
- call `listreceivedbyaddress()` filtering on `other_addr`. Assert only one transaction is returned.
- call `listreceivedbyaddress()` with no filtering. Assert both transactions are returned.